ElastosDIDSDK/DIDDocument.swift

```diff
@@ -262,21 +283,35 @@ public class DIDDocument {
         guard getMetadata().attachedStore else {
             throw DIDError.didStoreError("Not attached with DID store.")
         }
-        guard getMetadata().store!.containsPrivateKey(for: subject, id: ofId) else {
+        guard getMetadata().store!.containsPrivateKey(subject, ofId) else {
             throw DIDError.illegalArgument("Don't have private key")
         }
 
         let pubKey = publicKey(ofId: ofId)
         let pubs = pubKey!.publicKeyBytes
         let pubData = Data(bytes: pubs, count: pubs.count)
 
-        let privKey = try getMetadata().store?.loadPrivateKey(for: subject, byId: ofId)
-        let privKeyData = try DIDStore.decryptFromBase64(privKey!, storePassword)
-        let privateKeyData = try HDKey.DerivedKey.PEM_ReadPrivateKey(pubData, privKeyData)
+        let privKey = try getMetadata().store?.loadPrivateKey(subject, ofId, storePassword)
+        let privateKeyData = try HDKey.PEM_ReadPrivateKey(pubData, privKey!)
 
         return privateKeyData.data(using: .utf8)!
     }
 
+    // The result is extended private key format, the real private key is
+    // 32 bytes long start from position 46.
+    public func derive(_ index: Int, storepass: String) throws -> String {
+
+        guard !storepass.isEmpty else {
+            throw DIDError.illegalArgument("storepass is empty.")
+        }
+        guard getMetadata().attachedStore else {
+            throw DIDError.didStoreError("Not attached with a DID store.")
+        }
+
+        let key = HDKey.deserialize((try getMetadata().store?.loadPrivateKey(subject, getDefaultPublicKey()!, storepass))!)
```

Comment:
(try getMetadata().store?.loadPrivateKey(subject, getDefaultPublicKey()!, storepass))! 
too many force operator "!" , can we remove any of these?